application/authoriser/handler.py

```diff
@@ -24,18 +24,37 @@ def authorize_api_request(event, context) -> dict:
     return response
 
 def check_authorisation_token(token_hash_sent: str, noauth: bool) -> bool:
-    client = boto3.client("secretsmanager")
     if noauth and os.environ.get("DRD_ALLOW_NO_AUTH", "False") == "True":
         logger.log_info("Noauth actioned as allowed")
         return True
+
+    time_x = str(int((time.time() - 900) / 900))
+    time_y = str(int(time.time() / 900))
+    token_hash_sent_encoded = re.sub(r"^\$2y", "$2b", token_hash_sent).encode("utf-8")
+
+    if os.environ.get("RD_API_TOKEN_X_SLOT", "") == time_x and os.environ.get("RD_API_TOKEN_Y_SLOT", "") == time_y:
+        if os.environ.get("RD_API_TOKEN_X_TOKEN", "") == token_hash_sent or os.environ.get("RD_API_TOKEN_Y_TOKEN", "") == token_hash_sent:
```

Comment:
Needs to be encoded hash else it won't ever match

Reply:
..but we can't store bytes type in env?